addons/voice/org.openhab.voice.picotts/src/main/java/org/openhab/voice/picotts/internal/PicoTTSAudioStream.java

```diff
@@ -0,0 +1,125 @@
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.voice.picotts.internal;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+
+import org.eclipse.smarthome.core.audio.AudioException;
+import org.eclipse.smarthome.core.audio.AudioFormat;
+import org.eclipse.smarthome.core.audio.AudioStream;
+import org.eclipse.smarthome.core.audio.FixedLengthAudioStream;
+import org.eclipse.smarthome.core.voice.Voice;
+
+/**
+ * Implementation of {@link AudioStream} for {@link PicoTTSService}
+ *
+ * @author Florian Schmidt - Initial Contribution
+ */
+class PicoTTSAudioStream extends FixedLengthAudioStream {
+    private final Voice voice;
+    private final String text;
+    private final AudioFormat audioFormat;
+    private final InputStream inputStream;
+
+    private long length;
+    private File file;
+
+    public PicoTTSAudioStream(String text, Voice voice, AudioFormat audioFormat) throws AudioException {
+        this.text = text;
+        this.voice = voice;
+        this.audioFormat = audioFormat;
+        this.inputStream = createInputStream();
+    }
+
+    @Override
+    public AudioFormat getFormat() {
+        return audioFormat;
+    }
+
+    private InputStream createInputStream() throws AudioException {
+        String outputFile = generateOutputFilename();
+        String[] command = getCommand(outputFile);
+
+        try {
+            Process process = Runtime.getRuntime().exec(command);
+            process.waitFor();
+            file = new File(outputFile);
+            this.length = file.length();
+            return getFileInputStream(file);
+        } catch (IOException e) {
+            throw new AudioException("Error while executing '" + command + "'", e);
+        } catch (InterruptedException e) {
+            throw new AudioException("The '" + command + "' has been interrupted", e);
+        }
+    }
+
+    private InputStream getFileInputStream(File file) throws AudioException {
+        if (file == null) {
+            throw new IllegalArgumentException("file must not be null");
+        }
+        if (file.exists()) {
+            try {
+                return new FileInputStream(file);
+            } catch (FileNotFoundException e) {
+                throw new AudioException("Cannot open temporary audio file '" + file.getName() + ".");
+            }
+        } else {
+            throw new AudioException("Temporary file '" + file.getName() + "' not found!");
+        }
+    }
+
+    /**
+     * Generates a unique, absolute output filename
+     *
+     * @return Unique, absolute output filename
+     */
+    private String generateOutputFilename() throws AudioException {
+        File tempFile;
+        try {
+            tempFile = File.createTempFile(Integer.toString(text.hashCode()), ".wav");
+            tempFile.deleteOnExit();
```

Comment:
When the return statement is moved into this try clause the variable allocation can also be done inside which makes the code a bit more compact:

```java
File tempFile = File.createTempFile(Integer.toString(text.hashCode()), ".wav");
tempFile.deleteOnExit();
return tempFile.getAbsolutePath();
```

Reply:
Good point, I'll change that :)